library/components/rum/RumCrashReporterTask.bs

```diff
@@ -0,0 +1,141 @@
+sub init()
+    ddLogThread("RumCrashReporterTask.init()")
+    m.port = createObject("roMessagePort")
+    m.top.functionName = "sendCrashReport"
+end sub
+
+' ----------------------------------------------------------------
+' Reads the last known view event and send the crash report
+' ----------------------------------------------------------------
+sub sendCrashReport()
+    ddLogThread("RumCrashReporterTask.sendCrashReport()")
+    currentSessionlastViewEventFilePath = lastViewEventFilePath(m.top.instanceId)
+
+    unexpectedExitStatus = [
+        "EXIT_BRIGHTSCRIPT_CRASH",
+        "EXIT_BRIGHTSCRIPT_TIMEOUT",
+        "EXIT_GRAPHICS_NOT_RELEASED",
+        "EXIT_SIGNAL_TIMEOUT",
+        "EXIT_OUT_OF_MEMORY",
+        "EXIT_MEM_LIMIT_EXCEEDED_FG",
+        "EXIT_MEM_LIMIT_EXCEEDED_BG"
+    ]
+    keep = false
+    for each status in unexpectedExitStatus
+        if (m.top.lastExitOrTerminationReason = status)
+            keep = true
+        end if
+    end for
+
+    if (not keep)
+        ddLogInfo("Last exit status " + m.top.lastExitOrTerminationReason + " is not a crash, ignoring")
+    end if
+
+    folderPath = trackFolderPath("rum")
+    filenames = ListDir(folderPath)
+    for each filename in filenames
+        filePath = folderPath + "/" + filename
+        ddLogVerbose("RumCrashreporterTask, checking file: " + filename)
+        if (filePath = currentSessionlastViewEventFilePath)
+            ddLogVerbose("Ignoring current session file: " + filename)
+        else if (filename.Left(10) = "_last_view")
+            if (keep)
+                ddLogVerbose("Found last view file " + filePath)
+                sendCrashReportFromViewEventFile(filePath)
+            end if
+            ddLogVerbose("Deleting file " + filePath)
+            DeleteFile(filePath)
+        end if
+    end for
+
+    ddLogVerbose("RumCrashreporterTask, all checks done")
+end sub
+
+' ----------------------------------------------------------------
+' Sends a crash report linked to the view stored in the given path
+' @param filepath (string) the path to the file holding the last known view event
+' ----------------------------------------------------------------
+sub sendCrashReportFromViewEventFile(filepath as string)
+
+    ddLogVerbose("Reading file " + filepath)
+    lastViewEventJson = ReadAsciiFile(filepath)
+    if (lastViewEventJson = "" or lastViewEventJson = invalid)
+        ddLogWarning("A crash was detected but the last known view is empty, ignoring")
+        return
+    end if
+
+    ddLogVerbose("Parsing event from " + filepath)
+    lastViewEvent = ParseJson(lastViewEventJson)
+    if (lastViewEvent = invalid)
+        ddLogError("A crash was detected but the last view event is invalid")
+        ddLogError(lastViewEventJson)
+        return
+    end if
+
+    print "LastView:"; lastViewEvent
+    timeSpentNs& = lastViewEvent.view.time_spent + 100000000 ' 100 ms after last known timestamp
```

Comment:
Since you have millisToNanos, does it make senes to do this as `millisToNanos(100)` or is that just overhead that's not worth it?

Reply:
Indeed that's a good point